software/react-native/concepts.md

```diff
@@ -0,0 +1,60 @@
+# Basic Concepts
+
+#### Request / Response Data Flow
+
+In a standard redux project data flows like this:
+
+1. Dispatch an `attempting` action that will attempt to `POST`/`GET` data from a server. This action will usually flip an `attempting: true` flag in it's domain-related reducer in case we want to display some sort of UI to demonstrate to the user that a request is in process.
+
+2. A saga is listens for this `attempting` action, and then runs when it receives it along with it's params (*it will have passed through the reducer first*). The saga handles all side-effects associated with this action. The saga will be the place where the api call is actually made. Depending on it's success or failure a `success` or `failure` action will be dispatched to the reducer. If it's a `success` we dispatch the received data as well, if it's a `failure`, we display the received error messages.
+
+3. After the `success` action is dispatched from the reducer, the data will get stored as is or [normalized](https://github.com/paularmstrong/normalizr) and stored. We try to normalize all deeply nested data to make it much easier to work with. The `success` or `failure` action will also flip that `attempting: true` flag mentioned above back to `attempting: false` to show the user that the request has completed.
```

Comment:
are there any cases where normalizing it would be not the correct way to go? maybe not?